### PR TITLE
original_request_type for appeals & legacy appeals

### PR DESCRIPTION
**Pull request by tomas-nava**
Connects [CASEFLOW-425](https://vajira.max.gov/browse/CASEFLOW-425)

🥞  stacked PR 2 of 3
part 1: #15847
part 3: #15883

### Description
Adds a new column to appeals and legacy appeals that will store the hearing request type before any changes are made in Caseflow.

### Database Changes
*Only for Schema Changes*

* [x] Column comments updated
* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [x] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -1507,6 +1509,7 @@
     t.string "closest_regional_office"
     t.datetime "created_at"
     t.date "date_of_death", comment: "Date of Death reported by BGS, cached locally"
+    t.datetime "date_of_death_reported_at", comment: "The datetime that date_of_death last changed for veteran."
```
Comment (yoomlam) [inline db/schema.rb:1512]: How did this get here?

File: db/migrate/20210205182635_add_original_request_type_to_appeal.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class AddOriginalRequestTypeToAppeal < Caseflow::Migration
+  def change
+    add_column :appeals,
+      :original_request_type,
```
Comment (yoomlam) [inline db/migrate/20210205182635_add_original_request_type_to_appeal.rb:4]: Recommend incorporating `hearing` into the column name -- maybe `: orig_hearing_request_type`

Comment (yoomlam) [reply db/migrate/20210205182635_add_original_request_type_to_appeal.rb:4]: To be addressed in #15880

**Review by yoomlam — CHANGES_REQUESTED** [review]
Looks good! Need to remove unrelated changes from this PR.

File: docs/schema/caseflow.csv [hunk 3]
```diff
@@ -1143,6 +1145,7 @@ veterans,,,,,,,,
 veterans,closest_regional_office,string,,,,,,
 veterans,created_at,datetime,,,,,,
 veterans,date_of_death,date,,,,,,"Date of Death reported by BGS, cached locally"
+veterans,date_of_death_reported_at,datetime,,,,,,The datetime that date_of_death last changed for veteran.
```
Comment (yoomlam) [inline docs/schema/caseflow.csv:1148]:
```suggestion
```

File: db/schema.rb (hunk 1 above)
Comment (tomas-nava) [reply db/schema.rb]: good question. removed in d85e4e4

File: docs/schema/caseflow.csv (hunk 3 above)
Comment (tomas-nava) [reply docs/schema/caseflow.csv]: had to `make reset` and `make docs` to get rid of this one, committed in 6dd47fe

**Review by yoomlam — APPROVED** [review]
🎉

**Pull request merged**